pkg/providers/imagefamily/azlinux3.go

```diff
@@ -0,0 +1,135 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagefamily
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/bootstrap"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/customscriptsbootstrap"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/launchtemplate/parameters"
+
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+const (
+	AzureLinux3Gen2ImageDefinition    = "V3gen2"
+	AzureLinux3Gen1ImageDefinition    = "V3"
+	AzureLinux3Gen2ArmImageDefinition = "V3gen2arm64"
+)
+
+type AzureLinux3 struct {
+	Options *parameters.StaticParameters
+}
+
+func (u AzureLinux3) Name() string {
+	return v1beta1.AzureLinuxImageFamily
+}
+
+func (u AzureLinux3) DefaultImages() []DefaultImageOutput {
```

Comment:
Is anything different here, between v2, and v3 beyond `DefaultImages()`?

If we don't expect the bootstrapping, or naming to change, it might possibly make more sense for the DefaultImages to have `k8s` as a parameter?

Reply:
No, there are no other differences - _for now_. Agree it can be handled differently, and I think I tried that - and having a separate family was cleaner / more straightforward. Plus, right now Azure Linux is the only family where images depend on Kubernetes version, so would be good to have more examples before changing the pattern.

Reply:
I'm a little torn here, as I'm still somewhat leaning towards this being a combined family, as we are currently handling AzureLinux generically, and if we had a generic Ubuntu family, than I think we'd want to handle it the same way, and need to differentiate based on k8s. 

I think this split handling since one of our image families is generic and the other is versioned makes things a bit strange.

Its behind the scenes, so can be refactored later. Non-blocking, and we can always decide to adapt as we support other image families as well. I just expect if we add the generic option of Ubuntu as well, we'll want a combined overarching structure of it, to allow for reuse of bootstrapping, etc.

Reply:
Good points, let's revisit when we add more image families / options.